agent/rpc/peering/stream_test.go

```diff
@@ -24,6 +24,50 @@ import (
 	"github.com/hashicorp/consul/sdk/testutil/retry"
 )
 
+func TestStreamResources_Server_Follower(t *testing.T) {
+	publisher := stream.NewEventPublisher(10 * time.Second)
+	store := newStateStore(t, publisher)
+
+	srv := NewService(testutil.Logger(t), &testStreamBackend{
+		store: store,
+		pub:   publisher,
+		leader: func() bool {
+			return false
+		},
+	})
+
+	client := NewMockClient(context.Background())
+
+	errCh := make(chan error, 1)
+	client.ErrCh = errCh
+
+	go func() {
+		// Pass errors from server handler into ErrCh so that they can be seen by the client on Recv().
+		// This matches gRPC's behavior when an error is returned by a server.
+		err := srv.StreamResources(client.ReplicationStream)
+		if err != nil {
+			errCh <- err
+		}
+	}()
+	input := &pbpeering.ReplicationMessage{
+		Payload: &pbpeering.ReplicationMessage_Response_{
+			Response: &pbpeering.ReplicationMessage_Response{
+				ResourceURL: pbpeering.TypeURLService,
+				ResourceID:  "api-service",
+				Nonce:       "2",
+			},
+		},
+	}
+
+	err := client.Send(input)
+	require.NoError(t, err)
```

Comment:
I'm not sure if these lines are valuable; you're testing a `MockClient.Send` which will never return errors. We don't need to send a message for `StreamResources` to return an error.

Reply:
You are correct about the current "start of stream" leader check. If he were to add the "during a stream" leader check too, then it would need to be done using a Send.